common/src/main/java/tachyon/security/authentication/AuthenticationFactory.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security.authentication;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+import javax.security.sasl.SaslException;
+
+import org.apache.thrift.transport.TFramedTransport;
+import org.apache.thrift.transport.TSocket;
+import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportFactory;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.security.LoginUser;
+import tachyon.util.network.NetworkAddressUtils;
+
+/**
+ * This class is the main entry for Tachyon authentication. It switches different modes based on
+ * configuration, and provides corresponding Thrift class for authenticated connection between
+ * Client and Server.
+ */
+public final class AuthenticationFactory {
```

Comment:
I am a little confused with the naming of AuthenticationFactory and AuthenticationProviderFactory. What is the difference?

This class seems like it is only creating thrift servers and clients?